iree/hal/cuda/dynamic_symbols.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/hal/cuda/dynamic_symbols.h"
+
+#include <cstddef>
+
+#include "absl/base/attributes.h"
```

Comment:
We're trying to excise abseil as a dependency for non-test code, so whenever possible it's best to avoid using it (as otherwise it's just more code that needs to be ripped up later). Some places still need it (like the DynamicLibrary type) but those will be fixed soon - fine for example to continue passing the absl::Span today as it needs it, but in general it's best to treat any abseil use as tech debt. Here, for example, you shouldn't need attributes, memory, or str_cat - just span.

Reply:
Oops, yes sounds good. I just left dependency to span